### DISCUSSION TITLE
feature: Add Swift/SourceKit Integration

### DISCUSSION THREAD

**Discussion by Innei** [LazyVim/LazyVim#1380] [Q&A]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

Add Swift/SourceKit Integration, to make LazyVim better. There are lots of language supports, but I can not find Swift Support. I use Xcode to write swift code, but Xcode is too slow and has no copilot support.

### Describe the solution you'd like

Add Swift Integration, like TypeSciprt or another. Out of the box. Thank you!

### Describe alternatives you've considered

I found `xbase-lab/xbase` to it not integrate lsp keymap. And may be should manually configure sourcekit lsp.

**Comment by andreacfromtheapp** [comment]
I'm learning Swift and I'm long time vim/nvim user. I would love to see official support/integration into lazyvim. +1

Here's a recent blog post that shows how to do it on vanilla nvim. It could be a good starting point: https://wojciechkulik.pl/ios/the-complete-guide-to-ios-macos-development-in-neovim

I'm new to lazyvim (like since two days) and I'm struggling with extra configs and to implement it myeslf.

**Reply by Woit** [reply]
After JetBrains sunsetted AppCode, there is no way to normally do something for iOS/MacOS (because Xcode is crap). So LazyVim community can attract a lot of new developers, if swift-related projects will supported.

If someone (like me) lookign for solution, magic guy @wojciech-kulik wrote amazing article [here](https://wojciechkulik.pl/ios/the-complete-guide-to-ios-macos-development-in-neovim)

**Comment by lbennett-stacki** [comment]
I've started on a contribution here https://github.com/LazyVim/LazyVim/pull/3131

**Reply by dusty-phillips** [reply]
Awesome, thank you!

Not related to your work (probably), but I am getting a noticeable lag when I edit swift code in Neovim. Noticeable enough to be aggravating.

After a bit of hunting, I determined that it is the treesitter grammar that causes lag; `TSRemove swift` makes the lag go away.

Found https://github.com/alex-pinkus/tree-sitter-swift/issues/240 that may be cause-related. Unlike one commenter in that thread, `:TSBufDisable indent` did not help for me.

I'm evaluating Swift for a new project right now and trying to decide if this is a dealbreaker.

**Comment by antoniorodr** [comment]
How its going with this?

**Reply by nicholasss** [reply]
I am also wondering this. I would love to be able to work on Swift in LazyVim.

**Comment by raulp25** [comment]
Checking again

**Comment by zeevenn** [comment]
waiting

**Comment by erlangparasu** [comment]
any updates?
